src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -416,18 +410,25 @@ void AppHost::_HandleCreateWindow(const HWND hwnd, RECT proposedRect, LaunchMode
         const til::size availableSpace = desktopDimensions + nonClientSize;
 
         origin = til::point{
-            nearestMonitorInfo.rcWork.left - (nonClientSize.width() / 2),
-            nearestMonitorInfo.rcWork.top
+            ::base::saturated_cast<ptrdiff_t>(nearestMonitorInfo.rcWork.left - (nonClientSize.width() / 2)),

```

Comment:
I feel like this is weird. We're doing the math unprotected from over/underflow first then casting the result into `ptrdiff_t`.

Shouldn't we be doing a saturated operation on these things or casting them into the math wrappers first and then let it do the saturated subtract? 

The same goes for some of the other saturated casts below. Though I'm not totally clear on what the source types are here that requires us to do the cast to begin with.

Reply:
Yea you know, I felt weird writing that. I thinkthe types are LONGs originally. Maybe `point` is more forgiving than `size`, which is why I was getting weird "can't use an initializer list" errors. 

lemme back that out and spend more than 30 seconds on it